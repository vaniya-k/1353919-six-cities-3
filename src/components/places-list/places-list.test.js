import React from 'react';
import renderer from 'react-test-renderer';
import PlacesList from './places-list.jsx';
import placesListing from '../../mocks/places-listing-test.js';

it(`<PlaceList/>  with 54 matches and 3 places displayed`, () => {
  const mock = jest.fn();

  const tree = renderer
    .create(<PlacesList places={placesListing.places} foundPlacesQnt={placesListing.foundPlacesQnt} onPlaceCardClick={mock}/>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
