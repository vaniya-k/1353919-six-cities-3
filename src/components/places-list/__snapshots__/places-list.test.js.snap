// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlaceList/>  with 54 matches and 3 places displayed 1`] = `
<section
  className="cities__places places"
>
  <h2
    className="visually-hidden"
  >
    Places
  </h2>
  <b
    className="places__found"
  >
    54
     places to stay in Amsterdam
  </b>
  <form
    action="#"
    className="places__sorting"
    method="get"
  >
    <span
      className="places__sorting-caption"
    >
      Sort by
    </span>
    <span
      className="places__sorting-type"
      tabIndex="0"
    >
      Popular
      <svg
        className="places__sorting-arrow"
        height="4"
        width="7"
      >
        <use
          href="#icon-arrow-select"
        />
      </svg>
    </span>
    <ul
      className="places__options places__options--custom places__options--opened"
    >
      <li
        className="places__option places__option--active"
        tabIndex="0"
      >
        Popular
      </li>
      <li
        className="places__option"
        tabIndex="0"
      >
        Price: low to high
      </li>
      <li
        className="places__option"
        tabIndex="0"
      >
        Price: high to low
      </li>
      <li
        className="places__option"
        tabIndex="0"
      >
        Top rated first
      </li>
    </ul>
  </form>
  <div
    className="cities__places-list places__list tabs__content"
  >
    <article
      className="cities__place-card place-card"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="cities__image-wrapper place-card__image-wrapper"
      >
        <a
          href="#"
        >
          <img
            alt="Place image"
            className="place-card__image"
            height="200"
            src="img/room.jpg"
            width="260"
          />
        </a>
      </div>
      <div
        className="place-card__info"
      >
        <div
          className="place-card__price-wrapper"
        >
          <div
            className="place-card__price"
          >
            <b
              className="place-card__price-value"
            >
              €
              70
            </b>
            <span
              className="place-card__price-text"
            >
              / night
            </span>
          </div>
          <button
            className="place-card__bookmark-button place-card__bookmark-button--active button"
            type="button"
          >
            <svg
              className="place-card__bookmark-icon"
              height="19"
              width="18"
            >
              <use
                href="#icon-bookmark"
              />
            </svg>
            <span
              className="visually-hidden"
            >
              In bookmarks
            </span>
          </button>
        </div>
        <div
          className="place-card__rating rating"
        >
          <div
            className="place-card__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "80%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <h2
          className="place-card__name"
        >
          <a
            href="#"
          >
            Tiny apartment
          </a>
        </h2>
        <p
          className="place-card__type"
        >
          Apartment
        </p>
      </div>
    </article>
    <article
      className="cities__place-card place-card"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="cities__image-wrapper place-card__image-wrapper"
      >
        <a
          href="#"
        >
          <img
            alt="Place image"
            className="place-card__image"
            height="200"
            src="img/room.jpg"
            width="260"
          />
        </a>
      </div>
      <div
        className="place-card__info"
      >
        <div
          className="place-card__price-wrapper"
        >
          <div
            className="place-card__price"
          >
            <b
              className="place-card__price-value"
            >
              €
              160
            </b>
            <span
              className="place-card__price-text"
            >
              / night
            </span>
          </div>
          <button
            className="place-card__bookmark-button place-card__bookmark-button--active button"
            type="button"
          >
            <svg
              className="place-card__bookmark-icon"
              height="19"
              width="18"
            >
              <use
                href="#icon-bookmark"
              />
            </svg>
            <span
              className="visually-hidden"
            >
              In bookmarks
            </span>
          </button>
        </div>
        <div
          className="place-card__rating rating"
        >
          <div
            className="place-card__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "60%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <h2
          className="place-card__name"
        >
          <a
            href="#"
          >
            Spacious loft
          </a>
        </h2>
        <p
          className="place-card__type"
        >
          Apartment
        </p>
      </div>
    </article>
    <article
      className="cities__place-card place-card"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="cities__image-wrapper place-card__image-wrapper"
      >
        <a
          href="#"
        >
          <img
            alt="Place image"
            className="place-card__image"
            height="200"
            src="img/room.jpg"
            width="260"
          />
        </a>
      </div>
      <div
        className="place-card__info"
      >
        <div
          className="place-card__price-wrapper"
        >
          <div
            className="place-card__price"
          >
            <b
              className="place-card__price-value"
            >
              €
              220
            </b>
            <span
              className="place-card__price-text"
            >
              / night
            </span>
          </div>
          <button
            className="place-card__bookmark-button place-card__bookmark-button--active button"
            type="button"
          >
            <svg
              className="place-card__bookmark-icon"
              height="19"
              width="18"
            >
              <use
                href="#icon-bookmark"
              />
            </svg>
            <span
              className="visually-hidden"
            >
              In bookmarks
            </span>
          </button>
        </div>
        <div
          className="place-card__rating rating"
        >
          <div
            className="place-card__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "100%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <h2
          className="place-card__name"
        >
          <a
            href="#"
          >
            Rustic mansion
          </a>
        </h2>
        <p
          className="place-card__type"
        >
          House
        </p>
      </div>
    </article>
  </div>
</section>
`;
